### PR TITLE
Fix: Removed associated item gets corrected on story correction. [SDESK-4944]

### PR DESCRIPTION
**Pull request by GyanP**

### CODE REVIEW

File: apps/publish/content/common.py [hunk 1]
```diff
@@ -594,6 +594,12 @@ def _publish_associated_items(self, original, updates=None):
                     original[ASSOCIATIONS][associations_key]['operation'] = self.publish_type
                     continue
 
+                # Do not publish removed association items on correction.
```
Comment (ride90) [inline apps/publish/content/common.py]: I think you can simplify it with:
```
if updates.get(ASSOCIATIONS, {}).get(associations_key) is None:
    continue
```

File: apps/publish/content/common.py [hunk 2]
```diff
@@ -594,6 +594,12 @@ def _publish_associated_items(self, original, updates=None):
...
+                if (updates.get(ASSOCIATIONS, None)
+                        and associations_key in updates[ASSOCIATIONS]
+                        and updates[ASSOCIATIONS][associations_key] is None):
+                    continue
```
Comment (ride90) [inline apps/publish/content/common.py]: I think you can simplify it with:
```
if updates.get(ASSOCIATIONS, {}).get(associations_key) is None:
    continue
```

Comment (GyanP) [reply apps/publish/content/common.py]: Actually, association_key must be present, so check for updates[ASSOCIATIONS][association_key] is None only if association_key is present in updates.

Comment (ride90) [reply apps/publish/content/common.py]: makes sense

**Pull request merged**